src/Microsoft.AspNet.OData/Results/CreatedODataResult.cs

```diff
@@ -142,7 +144,7 @@ internal Uri EntityId
 
         internal IHttpActionResult GetInnerActionResult()
         {
-            if (RequestPreferenceHelpers.RequestPrefersReturnNoContent(Request))
+            if (RequestPreferenceHelpers.RequestPrefersReturnNoContent(new WebApiRequestHeaders(Request.Headers)))
```

Comment:
For debugging purposes, I would save `new WebApiRequestHeaders(Request.Headers)` into a `WebApiRequestHeaders` object and then pass the object to `RequestPrefersReturnNoContent`.